src/Compiler.Dynamic/StaticSystemWebCompilation.cs

```diff
@@ -39,36 +38,45 @@ internal sealed class StaticSystemWebCompilation : SystemWebCompilation<Persiste
         CancellationToken token)
     {
         // TODO: Handle output better
-        using var peStream = File.OpenWrite(Path.Combine(_options.Value.TargetDirectory, $"WebForms.{typeName}.dll"));
-        using var pdbStream = File.OpenWrite(Path.Combine(_options.Value.TargetDirectory, $"WebForms.{typeName}.pdb"));
-
-        peStream.SetLength(0);
-        pdbStream.SetLength(0);
+        using (var peStream = File.OpenWrite(Path.Combine(_options.Value.TargetDirectory, $"WebForms.{typeName}.dll")))
+        {
+            using var pdbStream = File.OpenWrite(Path.Combine(_options.Value.TargetDirectory, $"WebForms.{typeName}.pdb"));
+            peStream.SetLength(0);
+            pdbStream.SetLength(0);
 
-        var result = compilation.Emit(
-            embeddedTexts: embedded,
-            peStream: peStream,
-            pdbStream: pdbStream,
-            cancellationToken: token);
+            var result = compilation.Emit(
+                embeddedTexts: embedded,
+                peStream: peStream,
+                pdbStream: pdbStream,
+                cancellationToken: token);
 
-        if (!result.Success)
-        {
-            _logger.LogError("{ErrorCount} error(s) found compiling {Route}", result.Diagnostics.Length, route);
+            if (!result.Success)
+            {
+                _logger.LogError("{ErrorCount} error(s) found compiling {Route}", result.Diagnostics.Length, route);
 
-            var errors = GetErrors(result.Diagnostics).ToList();
-            var errorResult = JsonSerializer.Serialize(errors);
+                var errors = GetErrors(result.Diagnostics).ToList();
+                var errorResult = JsonSerializer.Serialize(errors);
 
-            File.WriteAllText(Path.Combine(_options.Value.TargetDirectory, $"{typeName}.errors.json"), errorResult);
+                File.WriteAllText(Path.Combine(_options.Value.TargetDirectory, $"{typeName}.errors.json"), errorResult);
 
-            throw new RoslynCompilationException(route, errors);
+                throw new RoslynCompilationException(route, errors);
+            }
         }
 
-        return new PersistedCompiledPage(new(route), embedded.Select(t => t.FilePath).ToArray())
+        var assembly = Assembly.LoadFile(Path.Combine(_options.Value.TargetDirectory, $"WebForms.{typeName}.dll"));
```

Comment:
I'm not happy with calling Assembly.LoadFile, but at least it's in the appropriate class. Have you figured out if we need to be able to activate the type or if it is only for metadata?

For now, we can leave it but please leave a //TODO in the code and create an issue to track replacing it

Reply:
I'll try to explain what we should probably do in the tracking issue, but for now, if it's working let's go forward with it

Reply:
#231 for us to traack